### PR TITLE
New branch

### PR DESCRIPTION
**Pull request by ClintonWeeYuan**

### CODE REVIEW

File: matchmake.js [hunk 1]
```diff
@@ -13,6 +13,32 @@ const criteria = {
...
+    "undergradField": "Computer Science"
+};
+
 function computeScore(applicant, advisor) {
-    //TODO code your function here
-}
\ No newline at end of file
+    let score = 0;
+    const applicant_keys = Object.keys(applicant);
+    const advisor_keys = Object.keys(advisor);
```
Comment (jonathanmcc) [inline matchmake.js:29]: advisor_keys was declared but never used

File: matchmake.js [hunk 2]
```diff
@@ -13,6 +13,32 @@ const criteria = {
...
+
+    for (let key of applicant_keys){
+      if (applicant[key] === advisor[key]){
```
Comment (jonathanmcc) [inline matchmake.js:32]: With this logic, if an applicant has `{"postgradUni":"cambridge"}` and the advisor has `{"postgradUni":"cambridge","postgradUniPrev":"cambridge"}`, the match will only gain score once. The correct result should gain the matchmaking score twice.

**Review by jonathanmcc — CHANGES_REQUESTED** [review]

File: matchmake.js (hunk 1 above)
Comment (ClintonWeeYuan) [reply matchmake.js]: Noted with thanks! I realised I didn't need it halfway through coding, but forgot to remove the variable

File: matchmake.js (hunk 2 above)
Comment (ClintonWeeYuan) [reply matchmake.js:31]: Oh, I forgot about that situation. Thanks for pointing it out.

**Pull request merged**